resources/osx/preinstall-extras

```diff
@@ -0,0 +1,16 @@
+## This works around an issue with the OSX service names in
+## puppet-agent. We're using a short service name intead of
+## the fully qualified name (puppet instead of com.puppetlabs.puppet)
+
+## Putting this workaround here until we can fix the service names in
+## the next major version, at which point the more generic logic in
+## vanagon will do the right thing
+
+if [ -n "$foundpkg" ]; then
```

Comment:
I generally prefer the form:

```bash
if [[ ${foundpkg:-UNDEFINED} == UNDEFINED ]]; then
```

I concede that this is not portable, and not intuitive if you're not deep in the bash minutia. I like it because it's extremely explicit. But the [Bash Pitfalls wiki entry has something to offer here](http://mywiki.wooledge.org/BashPitfalls#A.5B_-n_.24foo_.5D_or_.5B_-z_.24foo_.5D). It boils down to "it doesn't matter which form you use, as long as you remember to quote the variable"